File: Kotlin/BlackJack/src/unitTests/ConsoleInputHandlerTest.kt
package unitTests

import BlackJack.input.ConsoleInputHandler
import BlackJack.type.ActionType
import org.junit.Test

import org.junit.Assert.*
import java.io.ByteArrayInputStream

class ConsoleInputHandlerTest {

    private val cInputHandler = ConsoleInputHandler()

    @Test
    fun readNumberTest() {

        System.setIn(ByteArrayInputStream("5\n2.0\n2.2.2\n \n5a6".toByteArray()))

        assertEquals(5.0, cInputHandler.readNumber())

        assertEquals(2.0, cInputHandler.readNumber())

        assertEquals(null, cInputHandler.readNumber())

        assertEquals(null, cInputHandler.readNumber())

        assertEquals(null, cInputHandler.readNumber())

    }

    @Test
    fun readKey() {

        System.setIn(ByteArrayInputStream("s\nd\nq\nh\ne\nn\ni\n \nrossz input".toByteArray()))

        assertEquals(ActionType.SPLIT, cInputHandler.readKey())
        assertEquals(ActionType.DOUBLE, cInputHandler.readKey())
        assertEquals(ActionType.END, cInputHandler.readKey())
        assertEquals(ActionType.HIT, cInputHandler.readKey())
        assertEquals(ActionType.STAND, cInputHandler.readKey())
        assertEquals(ActionType.NEW, cInputHandler.readKey())
        assertEquals(ActionType.NEW, cInputHandler.readKey())
        assertEquals(ActionType.ERROR, cInputHandler.readKey())
        assertEquals(ActionType.ERROR, cInputHandler.readKey())

    }
}

File: Lua/BlackJack/blackjack/test/consoleInputHandlerTest.lua

h = require "test.testHelper"
io.read = h.read

local ActionType = require "type.ActionType"
local ConsoleInputHandler = require("input.ConsoleInputHandler"):getInstance()


function testReadNumber()

    h.input = {"5", "2.0", "2.2.2", " ", "5a6"}
    h.cnt = 0

    lu.assertEquals(ConsoleInputHandler.readNumber(), 5) -- A number típus igazából nem tesz különbséget 5 és 5.0 között
    
    lu.assertEquals(ConsoleInputHandler.readNumber(), 2.0)
    
    lu.assertEquals(ConsoleInputHandler.readNumber(), nil)
    
    lu.assertEquals(ConsoleInputHandler.readNumber(), nil)
    
    lu.assertEquals(ConsoleInputHandler.readNumber(), nil)

end


function testReadKey()

    h.input = {"s", "d", "q", "h", "e", "n", "i", " ", "rossz input"}
    h.cnt = 0

    lu.assertEquals(ConsoleInputHandler.readKey(), ActionType.SPLIT)
    lu.assertEquals(ConsoleInputHandler.readKey(), ActionType.DOUBLE)
    lu.assertEquals(ConsoleInputHandler.readKey(), ActionType.END)
    lu.assertEquals(ConsoleInputHandler.readKey(), ActionType.HIT)
    lu.assertEquals(ConsoleInputHandler.readKey(), ActionType.STAND)
    lu.assertEquals(ConsoleInputHandler.readKey(), ActionType.NEW)
    lu.assertEquals(ConsoleInputHandler.readKey(), ActionType.NEW)
    lu.assertEquals(ConsoleInputHandler.readKey(), ActionType.ERROR)
    lu.assertEquals(ConsoleInputHandler.readKey(), ActionType.ERROR)

end

